### PR TITLE
add animation to directory change

### PR DESCRIPTION
**Pull request by theScrabi**
Implements #474

### CODE REVIEW

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 1]
```diff
@@ -657,14 +659,70 @@ public int onBrowseUp() {
...
+            @Override
+            public void onAnimationStart(Animation animation) {
+
+            }
+
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                listDirectory(file/*, MainApp.getOnlyOnDevice()*/);
+                Animation fadeInBack =
+                        AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadein_back);
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: This line fits properly in the previous one, you do not need to use a line break

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 2]
```diff
@@ -657,14 +659,70 @@ public int onBrowseUp() {
...
+            }
+        });
+        getListView().startAnimation(fadeOutFront);
+    }
+
+    private void listDirectoryWidthAnimationUp(final OCFile file) {
+        if(getListView().getVisibility() == View.GONE) {
+            listDirectory(file);
+            Animation fadeInFront =
+                    AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadein_front);
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: Same here

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 3]
```diff
@@ -657,14 +659,70 @@ public int onBrowseUp() {
...
+            @Override
+            public void onAnimationStart(Animation animation) {
+
+            }
+
+            @Override
+            public void onAnimationEnd(Animation animation) {
+                listDirectory(file);
+                Animation fadeInFront =
+                        AnimationUtils.loadAnimation(getContext(), R.anim.dir_fadein_front);
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: Same here

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 4]
```diff
@@ -657,14 +659,70 @@ public int onBrowseUp() {
...
+
     @Override
     public void onItemClick(AdapterView<?> l, View v, int position, long id) {
         OCFile file = (OCFile) mAdapter.getItem(position);
         if (file != null) {
             if (file.isFolder()) {
                 // update state and view of this fragment
                 // TODO Enable when "On Device" is recovered ?
-                listDirectory(file/*, MainApp.getOnlyOnDevice()*/);
+                listDirectoryWithAnimationDown(file  /*, MainApp.getOnlyOnDevice()*/);
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: Do we need the second parameter `MainApp.getOnlyOnDevice()` or we can just delete it?

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java [hunk 5]
```diff
@@ -44,6 +44,8 @@
 import android.view.View;
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: Please update the copyright year to 2018 and include yourself as @author

File: src/com/owncloud/android/ui/fragment/FileFragment.java [hunk 6]
```diff
@@ -144,7 +144,7 @@ public void onSyncEvent(String syncEvent, boolean success, OCFile updatedFile) {
          * 
          * @param file      File to show details
          */
-        public void showDetails(OCFile file);
+        void showDetails(OCFile file);
```
Comment (davigonz) [inline src/com/owncloud/android/ui/fragment/FileFragment.java]: Please update the copyright year to 2018 and include yourself as @author at the top of the class

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 4 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: I'm not sure, I don't know what MainApp.getOnlyOnDevice() should do. I't is not defined however.

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 5 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: done

File: src/com/owncloud/android/ui/fragment/FileFragment.java (hunk 6 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/fragment/FileFragment.java]: done

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 3 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: done

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 2 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: done

File: src/com/owncloud/android/ui/fragment/OCFileListFragment.java (hunk 1 above)
Comment (theScrabi) [reply src/com/owncloud/android/ui/fragment/OCFileListFragment.java]: done

**Comment by davigonz** [comment]
Think this issue has not passed QA @theScrabi 😬 , @jesmrec has approved the code, but not executed the corresponding tests.

**Comment by jesmrec** [comment]
@theScrabi i have noticed the current behaviour related to this PR.

Check the following steps:

1. From root folder, browse into a folder that contains subfolders
2. Browse back to root
3. Browse again into the folder in step 1

Current behaviour: app crashes
Expected behaviour: app does not crash, and folder is browsed correctly

Tested with Nexus 6P v7

Stack trace:

E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: com.owncloud.android.debug, PID: 15977
                  java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
                      at java.util.ArrayList.get(ArrayList.java:411)
                      at android.widget.HeaderViewListAdapter.isEnabled(HeaderViewListAdapter.java:164)
                      at android.widget.ListView.dispatchDraw(ListView.java:3419)
                      at android.view.View.draw(View.java:17188)
                      at android.widget.AbsListView.draw(AbsListView.java:4315)
                      at android.view.View.updateDisplayListIfDirty(View.java:16167)
                      at android.view.View.draw(View.java:16951)
                      at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
                      at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
                      at android.view.View.draw(View.java:17188)
                      at android.view.View.updateDisplayListIfDirty(View.java:16167)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ViewGroup.recreateChildDisplayList(ViewGroup.java:3711)
                      at android.view.ViewGroup.dispatchGetDisplayList(ViewGroup.java:3691)
                      at android.view.View.updateDisplayListIfDirty(View.java:16130)
                      at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:648)
                      at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:654)
                      at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:762)
                      at android.view.ViewRootImpl.draw(ViewRootImpl.java:2800)
                      at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2608)
                      at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2215)
                      at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
                      at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6338)
                      at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
                      at android.view.Choreographer.doCallbacks(Choreographer.java:686)
E/AndroidRuntime:     at android.view.Choreographer.doFrame(Choreographer.java:621)
                      at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
                      at android.os.Handler.handleCallback(Handler.java:751)
                      at android.os.Handler.dispatchMessage(Handler.java:95)
                      at android.os.Looper.loop(Looper.java:154)
                      at android.app.ActivityThread.main(ActivityThread.java:6121)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)

**Comment by theScrabi** [comment]
@jesmrec This will provide a fix:
https://github.com/owncloud/android/pull/2154

**Pull request merged**